File: AILang/benchmarks/c/gauntlet_protocol_erased.c
#include <stdint.h>
#include <stdio.h>

static inline int64_t protocol_scan_erased(int64_t iterations) {
    const int64_t packet_hash = 393291961LL;
    int64_t acc = 0;
    for (int64_t i = 0; i < iterations; ++i) {
        acc = (acc + packet_hash + i) % 1000000007LL;
    }
    return acc;
}

int main(void) {
    printf("%lld\n", (long long)protocol_scan_erased(1200000LL));
    return 0;
}

File: AILang/benchmarks/rust/gauntlet_protocol_erased.rs
#[inline(always)]
fn protocol_scan_erased(iterations: i64) -> i64 {
    let packet_hash = 393_291_961_i64;
    let mut acc = 0_i64;
    let mut i = 0_i64;
    while i < iterations {
        acc = (acc + packet_hash + i) % 1_000_000_007;
        i += 1;
    }
    acc
}

fn main() {
    println!("{}", protocol_scan_erased(1_200_000));
}
